app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -4,23 +4,60 @@
 
 package org.mozilla.focus.browser
 
+import android.animation.AnimatorSet
+import android.animation.ObjectAnimator
 import android.content.Context
+import android.graphics.Color
 import android.graphics.drawable.Drawable
 import android.support.v4.content.ContextCompat
+import android.support.v7.widget.GridLayoutManager
+import android.support.v7.widget.RecyclerView
 import android.util.AttributeSet
+import android.util.TypedValue
 import android.view.LayoutInflater
 import android.view.View
+import android.view.ViewGroup
+import android.view.animation.DecelerateInterpolator
 import android.widget.LinearLayout
 import kotlinx.android.synthetic.main.browser_overlay.view.*
+import kotlinx.android.synthetic.main.home_tile.view.*
+import kotlinx.coroutines.experimental.CompletableDeferred
+import kotlinx.coroutines.experimental.CoroutineStart
+import kotlinx.coroutines.experimental.Job
+import kotlinx.coroutines.experimental.android.UI
+import kotlinx.coroutines.experimental.async
+import kotlinx.coroutines.experimental.launch
 import org.mozilla.focus.R
 import org.mozilla.focus.autocomplete.UrlAutoCompleteFilter
+import org.mozilla.focus.ext.forceExhaustive
+import org.mozilla.focus.ext.toJavaURI
+import org.mozilla.focus.ext.withRoundedCorners
+import org.mozilla.focus.home.BundledHomeTile
+import org.mozilla.focus.home.BundledTilesManager
+import org.mozilla.focus.home.CustomHomeTile
+import org.mozilla.focus.home.CustomTilesManager
+import org.mozilla.focus.home.HomeTile
+import org.mozilla.focus.home.HomeTilePlaceholderGenerator
+import org.mozilla.focus.home.HomeTileScreenshotStore
 import org.mozilla.focus.telemetry.TelemetryWrapper
+import org.mozilla.focus.utils.FormattedDomain
+import org.mozilla.focus.utils.OnUrlEnteredListener
 import org.mozilla.focus.utils.Settings
 import org.mozilla.focus.widget.InlineAutocompleteEditText
 
+/**
```

Comment:
\[detekt\] INFO: \[comments/CommentOverPrivateProperty\] Private properties should be named such that they explain themselves even without a comment\.